spdx-utils/src/main/kotlin/Extensions.kt

```diff
@@ -26,6 +26,37 @@ import java.util.EnumSet
  */
 operator fun <E : Enum<E>> EnumSet<E>.plus(other: EnumSet<E>): EnumSet<E> = EnumSet.copyOf(this).apply { addAll(other) }
 
+/**
+ * Check that the map to be created from the list of pairs would not contain any identical keys. Throws an
+ * [IllegalArgumentException] if such keys are found.
+ */
+fun <V> List<Pair<String, V>>.checkForSameKeyCaptialization(): Map<String, V> =
```

Comment:
This also converts the list of pairs into a map and returns it which is not obvious from the name. I would suggest to move the `toMap()` out of the function and call it after this check. I think this leads to more readable client code than trying to find a name that describes both actions, the conversion and the check.

Reply:
This became obsolete now.